src/main/java/backend/Logic.java

```diff
@@ -196,18 +191,36 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        eagerlyUpdateIssueLabels(issue, newLabels);
+
         logger.info("Changing labels for " + issue + " on GitHub");
-        return repoIO.replaceIssueLabels(issue, newLabels)
-                .thenApply(labels -> {
-                    logger.info("Changing labels for " + issue + " on UI");
-                    issue.setLabels(labels);
-                    refreshUI();
-                    return true;
-                })
+        return repoOpControl.replaceIssueLabels(issue, newLabels)
+                .thenApply(labels -> labels.containsAll(newLabels))
                 .exceptionally(Futures.withResult(false));
     }
 
     /**
+     * Assigns a list of labels to an issue offline and immediately reflect the change on the UI without
+     * concerning whether the labels have been updated on the remote repo's data source
+     *
+     * The UI will not be force refreshed if the set of labels are the same as the existing labels
+     * assigned to the issue.
```

Comment:
This sentence is not actually part of this method. i.e. the code doesn't match this description. Based on the code, it simply refreshes the UI. On a related note, I wonder if `refreshUI();` should be part of this method. The abstraction level doesn't seem right.